src/lualib/Decorator.d.ts

```diff
@@ -0,0 +1,5 @@
+export type Decorator<TTarget extends AnyTable, TKey extends keyof TTarget> = (
```

Comment:
Decorator sounds extremely generic, what is this used for?

Reply:
It's used in both Decorate.ts and DecorateParam.ts, which are independent lualib features.